src/web/components/StickyBottomBanner/BrazeBanner.tsx

```diff
@@ -114,57 +108,36 @@ const getMessageFromBraze = async (
 					undefined,
 					internalButtonId,
 				);
-				appboy.logInAppMessageButtonClick(thisButton, message);
+				appboy.logInAppMessageButtonClick(thisButton, message as any);
 			};
 
 			const logImpressionWithBraze = () => {
 				// Log the impression with Braze
 				appboy.logInAppMessageImpression(message);
 			};
-
-			if (extras) {
-				const meta = {
-					dataFromBraze: extras,
-					logImpressionWithBraze,
-					logButtonClickWithBraze,
-				};
-
-				resolve({ result: true, meta });
-			} else {
-				resolve({ result: false });
-			}
-
-			// Unsubscribe
-			if (subscriptionId) {
-				appboy.removeSubscription(subscriptionId);
-			}
-		};
-
-		// Keep hold of the subscription ID so that we can unsubscribe in the
-		// callback, ensuring that the callback is only invoked once per page
-		subscriptionId = appboy.subscribeToInAppMessage(callback);
-
-		setHasCurrentBrazeUser();
-		appboy.changeUser(brazeUuid);
-		appboy.openSession();
-	});
-
-	canShowPromise
-		.then(() => {
+			const meta = {
+				dataFromBraze: message.extras,
+				logImpressionWithBraze,
+				logButtonClickWithBraze,
+			};
+			return { result: true, meta };
+		})
+		.then((outcome) => {
 			const appboyTimeTaken = appboyTiming.end();
 
 			record({
 				component: 'braze-appboy-timing',
 				value: appboyTimeTaken,
 			});
+
```

Comment:
I realise this was here before, but should we call `appboyTiming.clear` from a finally block to make sure it gets called either way? It's not clear to me from the docs whether calling clear after you are done with the timer is required to:
* prevent the browser keeping its performance monitoring apparatus engaged
* make sure we don't accidentally use a previous measurement

Reply:
Hmm I don't remember the detail here. @rhystmills are you able to shed any light?

Reply:
Yes that makes more sense than it being in the catch, so I'll move it.

I was under the impression that performance marks were scoped to the page, but I'm also having a hard time finding conclusive documentation that states that. Perhaps this is one to look into for a future PR? It would apply wherever else `initPerf` is used across the website.